src/services/AssetService.ts

```diff
@@ -135,6 +157,65 @@ export class Assets {
             this.add(result);
         });
     }
+
+    public update(uuid: string, content: any): Promise<Asset> {
+        const result = new Promise<Asset>((resolve, reject) => {
+            if (this.localStorage) {
+                const asset = this.assets[uuid];
+                asset.content = content;
+                resolve(asset);
+            } else {
+                const url = `${this.endpoint}?uuid=${uuid}`;
+                axios
+                    .post(url, content)
+                    .then((response: AxiosResponse) => {
+                        const ob = response.data.results[0];
+                        const asset = {
+                            id: ob.uuid,
+                            name: ob.name,
+                            type: this.assetType,
+                            content: ob.definition
+                        };
+                        if (this.localStorage) {
+                            this.assets[uuid] = asset;
+                        }
+                        return resolve(asset);
+                    })
+                    .catch(error => reject(error));
+            }
+        });
+
+        return result;
+    }
+
+    public getAssetSet(): any[] {
+        const assets: any[] = [];
```

Comment:
small deal, but i think both `getAssetSet` and `getAssetContents` we should be making use of `Array.prototype.reduce` instead of a `map/push` combination. what do you think?

Reply:
Not sure how that would be nicer operating against a map. I think would agree if we could use `entries()` but since it's an array of keys, don't think reduce makes sense.

Reply:
Fair enough though. Not wild about this pattern either so I’ll play with it a bit first.

Reply:
Yeah not a big deal, FWIW was thinking:

```
return Object.keys(this.assets).reduce((assets, key) => {
    const { type, id, name, content } = this.assets[key];
    if (type !== AssetType.Property) {
        return [...assets, {
                [this.idProperty]: id,
                name,
                ...content
            }
		];
    }
    return assets;
}, []);
```

I think in the future a `forEach` would make more sense than `map` since `map` implies that a new array is being created.

Reply:
Right. That’s a lot of dots. :) Lemme play with it a bit. I’m heading out right now but can look later.

Reply:
Yeah yeah...😉 

`concat` would probably work as an alternative to array destructuring, i'm cool w/ a `forEach`/`push` combo tho 👍

Reply:
Switched this to use `map` properly (instead of `forEach`), which I believe is the correct functional way to do this.

Reply:
i think `reduce` is the ideal method here because it allows you to conditionally return a value to the accumulator. the problem w/ map is that it returns a value no matter what.

Reply:
sounds good, no strong feelings either way re: `reduce` vs `forEach`/`push`. also, my b, hadn't seen that the new implementation of `getAssetContents` doesn't have an `if` block; in that case, agree, `map` is the method.